go-controller/pkg/ovn/secondary_layer2_network_controller_test.go

```diff
@@ -575,3 +503,117 @@ func nodeCIDR() *net.IPNet {
 		Mask: net.CIDRMask(24, 32),
 	}
 }
+
+func setupFakeOvnForLayer2Topology(fakeOvn *FakeOVN, initialDB libovsdbtest.TestSetup, netInfo secondaryNetInfo, testNode *v1.Node, podInfo testPod, pod *v1.Pod) error {
+	var err error
+	By(fmt.Sprintf("creating a network attachment definition for network: %s", netInfo.netName))
+	nad, err := newNetworkAttachmentDefinition(
+		ns,
+		nadName,
+		*netInfo.netconf(),
+	)
+	if err != nil {
+		return err
+	}
+	By("setting up the OVN DB without any entities in it")
+	err = netInfo.setupOVNDependencies(&initialDB)
+	if err != nil {
+		return err
+	}
+
+	if netInfo.isPrimary {
+		networkConfig, err := util.NewNetInfo(netInfo.netconf())
+		if err != nil {
+			return err
+		}
+
+		initialDB.NBData = append(
+			initialDB.NBData,
+			&nbdb.LogicalRouter{
+				Name:        fmt.Sprintf("GR_%s_%s", networkConfig.GetNetworkName(), nodeName),
+				ExternalIDs: standardNonDefaultNetworkExtIDs(networkConfig),
+			},
+		)
+	}
+
+	fakeOvn.startWithDBSetup(
+		initialDB,
+		&v1.NamespaceList{
+			Items: []v1.Namespace{
+				*newNamespace(ns),
+			},
+		},
+		&v1.NodeList{Items: []v1.Node{*testNode}},
+		&v1.PodList{
+			Items: []v1.Pod{
+				*pod,
+			},
+		},
+		&nadapi.NetworkAttachmentDefinitionList{
+			Items: []nadapi.NetworkAttachmentDefinition{*nad},
+		},
+	)
+	podInfo.populateLogicalSwitchCache(fakeOvn)
+
+	// on IC, the test itself spits out the pod with the
+	// annotations set, since on production it would be the
+	// clustermanager to annotate the pod.
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		By("asserting the pod originally does *not* feature the OVN pod networks annotation")
+		// pod exists, networks annotations don't
+		pod, err := fakeOvn.fakeClient.KubeClient.CoreV1().Pods(podInfo.namespace).Get(context.Background(), podInfo.podName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		_, ok := pod.Annotations[util.OvnPodAnnotationName]
+		if ok {
+			return fmt.Errorf("expected pod annotation %q", util.OvnPodAnnotationName)
+		}
+	}
+	err = fakeOvn.controller.nadController.Start()
+	if err != nil {
+		return err
+	}
+
+	err = fakeOvn.controller.WatchNamespaces()
+	if err != nil {
+		return err
+	}
+	err = fakeOvn.controller.WatchPods()
+	if err != nil {
+		return err
+	}
+	By("asserting the pod (once reconciled) *features* the OVN pod networks annotation")
+	secondaryNetController, ok := fakeOvn.secondaryControllers[secondaryNetworkName]
+	if !ok {
```

Comment:
nit: let's use something more explicit than `ok` . `doesControllerExist` for instance.

Reply:
DONE